src/main/java/dansplugins/factionsystem/commands/JoinCommand.java

```diff
@@ -11,8 +12,63 @@
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
-public class JoinCommand {
+public class JoinCommand extends SubCommand {
 
+    public JoinCommand() {
+        super(new String[] {
+                "join", LOCALE_PREFIX + "CmdJoin"
+        }, true);
+    }
+
+    /**
+     * Method to execute the command for a player.
+     *
+     * @param player who sent the command.
+     * @param args   of the command.
+     * @param key    of the sub-command (e.g. Ally).
+     */
+    @Override
+    public void execute(Player player, String[] args, String key) {
+        final String permission = "mf.join";
+        if (!(checkPermissions(player, permission))) return;
+        if (args.length == 0) {
+            player.sendMessage(translate("&c" + getText("UsageJoin")));
+            return;
+        }
+        if (data.isInFaction(player.getUniqueId())) {
+            player.sendMessage(translate("&c" + getText("AlertAlreadyInFaction")));
+            return;
+        }
+        final Faction target = getFaction(String.join(" ", args));
+        if (target == null) {
+            player.sendMessage(translate("&c" + getText("FactionNotFound")));
+            return;
+        }
+        FactionJoinEvent joinEvent = new FactionJoinEvent(faction, player);
+        Bukkit.getPluginManager().callEvent(joinEvent);
+        if (joinEvent.isCancelled()) {
+            // TODO Locale Message
+            return;
+        }
+        faction.addMember(player.getUniqueId(), data.getPlayersPowerRecord(player.getUniqueId()).getPowerLevel());
+        faction.uninvite(player.getUniqueId());
+        messageFaction(faction, translate("&a" + getText("HasJoined", player.getName(), faction.getName())));
+        player.sendMessage(translate("&a" + getText("AlertJoinedFaction")));
+    }
+
+    /**
+     * Method to execute the command.
+     *
+     * @param sender who sent the command.
+     * @param args   of the command.
+     * @param key    of the command.
+     */
+    @Override
+    public void execute(CommandSender sender, String[] args, String key) {
+
```

Comment:
Are there plans for these empty methods? If, not perhaps a comment should be put stating that implementation of the method is optional or unnecessary, or maybe the empty methods should be removed. I don't mind if they stay in the implementation, though, honestly.